src/Features/CSharp/Portable/CSharpFeaturesResources.resx

```diff
@@ -461,4 +461,22 @@
   <data name="NamespaceName" xml:space="preserve">
     <value>&lt;namespace name&gt;</value>
   </data>
+  <data name="AutoselectDisabledDueToClassDeclaration" xml:space="preserve">
```

Comment:
So, IMO, i prefer to eschew excess code that has little marginal utility.  In this case it seems to be excess code that doesn't actually end up improving the user experience in any real way.  

It's similar to if we were to take every error that might be put on an an expression and were to specialize it for every type of expression.

I like localization specialization where there is something practically different that we'd be saying.  Honestly, for the entire class of messages that are nothing more than "completion is in builder mode because you could be declaring a variable" i would be fine with a single message.  I don't think there's value actually saying "you could be declaring a namespace" or a struct, or a class.  It's all clear from context.  About the only one that is slightly special is lambdas.  That's because there is no clear syntax when you're writing it that you're making a lambda.  And it can definitely be non-obvious why we picked the builder.

So, personally, i'd prefer this code to be removed as i don't see it justifying itself.